flink-streaming-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/SimpleConsumerThread.java

```diff
@@ -408,23 +412,32 @@ else if (partitionsRemoved) {
 		}
 	}
 
-	private void getMissingOffsetsFromKafka(
+	private void replaceEarliestOrLatestOffsetsWithActualValuesFromKafka(
 			List<KafkaTopicPartitionState<TopicAndPartition>> partitions) throws IOException
 	{
 		// collect which partitions we should fetch offsets for
-		List<KafkaTopicPartitionState<TopicAndPartition>> partitionsToGetOffsetsFor = new ArrayList<>();
+		List<KafkaTopicPartitionState<TopicAndPartition>> partitionsWithEarliestOffsetSetting = new ArrayList<>();
```

Comment:
Good point, didn't think of that. I'll call `getLastOffsetFromKafka` if getOffset() returns `OffsetRequest.EarliestTime()` or `OffsetRequest.LatestTime()`.